src/write_outputs/write_allamcyclelox.jl

```diff
@@ -0,0 +1,178 @@
+		dfCapAllam.EndCap_sCO2turbine_MW_gross = dfCapAllam.EndCap_sCO2turbine_MW_gross * ModelScalingFactor
+
+		dfCapAllam.StartCap_ASU_MW_gross = dfCapAllam.StartCap_ASU_MW_gross* ModelScalingFactor
+		dfCapAllam.RetCap_ASU_MW_gross = dfCapAllam.RetCap_ASU_MW_gross * ModelScalingFactor
+		dfCapAllam.NewCap_ASU_MW_gross = dfCapAllam.NewCap_ASU_MW_gross * ModelScalingFactor
+		dfCapAllam.EndCap_ASU_MW_gross = dfCapAllam.EndCap_ASU_MW_gross * ModelScalingFactor
+
+		dfCapAllam.StartCap_LOX_t = dfCapAllam.StartCap_LOX_t * ModelScalingFactor
+		dfCapAllam.RetCap_LOX_t = dfCapAllam.RetCap_LOX_t * ModelScalingFactor
+		dfCapAllam.NewCap_LOX_t = dfCapAllam.NewCap_LOX_t * ModelScalingFactor
+		dfCapAllam.EndCap_LOX_t = dfCapAllam.EndCap_LOX_t * ModelScalingFactor
+	end
+
+	total_allam = DataFrame(
+			Resource = "Total", Zone = "n/a", 
+			StartCap_sCO2turbine_MW_gross = sum(dfCapAllam[!,:StartCap_sCO2turbine_MW_gross]), 
+			RetCap_sCO2turbine_MW_gross = sum(dfCapAllam[!,:RetCap_sCO2turbine_MW_gross]),
+			NewCap_sCO2turbine_MW_gross = sum(dfCapAllam[!,:NewCap_sCO2turbine_MW_gross]), 
+			EndCap_sCO2turbine_MW_gross = sum(dfCapAllam[!,:EndCap_sCO2turbine_MW_gross]),
+
+			StartCap_ASU_MW_gross = sum(dfCapAllam[!,:StartCap_ASU_MW_gross]), 
+			RetCap_ASU_MW_gross = sum(dfCapAllam[!,:RetCap_ASU_MW_gross]),
+			NewCap_ASU_MW_gross = sum(dfCapAllam[!,:NewCap_ASU_MW_gross]), 
+			EndCap_ASU_MW_gross = sum(dfCapAllam[!,:EndCap_ASU_MW_gross]),
+
+			StartCap_LOX_t = sum(dfCapAllam[!,:StartCap_LOX_t]), 
+			RetCap_LOX_t = sum(dfCapAllam[!,:RetCap_LOX_t]),
+			NewCap_LOX_t = sum(dfCapAllam[!,:NewCap_LOX_t]), 
+			EndCap_LOX_t = sum(dfCapAllam[!,:EndCap_LOX_t]),
+		)
+
+	dfCapAllam = vcat(dfCapAllam, total_allam)
+	CSV.write(joinpath(path,"capacity_allam_cycle_lox.csv"), dfCapAllam)
+
+	# also write the vOutput_AllamcycleLOX, vLOX_in, vLOX_out
+	
+	
+end
+
+function write_allam_output(path::AbstractString, inputs::Dict, setup::Dict, EP::Model)
+	ALLAM_CYCLE_LOX = inputs["ALLAM_CYCLE_LOX"] 
+	T = inputs["T"]
+    # Allam cycle components
+    # by default, i = 1 -> sCO2Turbine; i = 2 -> ASU; i = 3 -> LOX
+    sco2turbine, asu, lox = 1, 2, 3
+
+	@expression(EP, eNetPowerAllam[y in ALLAM_CYCLE_LOX, t = 1:T],
+        EP[:eP_Allam][y,t] - EP[:vCHARGE_ALLAM][y,t])
+
+    # Power injected by each resource in each time step
+    dfAllam_output = DataFrame(Resource = 
+		[inputs["RESOURCE_NAMES"][ALLAM_CYCLE_LOX] .*"_sco2turbine_gross_power_mw";
+		inputs["RESOURCE_NAMES"][ALLAM_CYCLE_LOX] .*"_sco2turbine_commit";
+		inputs["RESOURCE_NAMES"][ALLAM_CYCLE_LOX] .*"_asu_gross_power_mw";
+		inputs["RESOURCE_NAMES"][ALLAM_CYCLE_LOX] .*"_asu_commit";
+		inputs["RESOURCE_NAMES"][ALLAM_CYCLE_LOX] .*"_net_power_output_mw";
+		inputs["RESOURCE_NAMES"][ALLAM_CYCLE_LOX] .*"_storage_lox_t";
+		inputs["RESOURCE_NAMES"][ALLAM_CYCLE_LOX] .*"_lox_in_t";
+		inputs["RESOURCE_NAMES"][ALLAM_CYCLE_LOX] .*"_lox_out_t";
+		inputs["RESOURCE_NAMES"][ALLAM_CYCLE_LOX] .*"_gox_t"])
```

Comment:
Consider assigning a variable name like `allam_resources = inputs["RESOURCE_NAMES"][ALLAM_CYCLE_LOX]` for brevity.